app/components/ReputationGuard.jsx

```diff
@@ -0,0 +1,23 @@
+import React from "react"
+import { connect } from "react-redux"
+import { translate } from 'react-i18next'
+import { isAuthenticated, hasReputation } from '../state/users/current_user/selectors'
+import { LoadingFrame, LinkWithIcon } from "./Utils"
+
+@connect((state, props) => ({
+  isAuthenticated: isAuthenticated(state),
+  hasReputation: hasReputation(state, props.requiredRep),
+  isLoading: state.CurrentUser.isLoading
+}))
+@translate('errors')
+export default class ReputationGuard extends React.PureComponent {
+  render() {
+    const { isLoading, hasReputation, t } = this.props
+
+    if (isLoading)
+      return <LoadingFrame/>
```

Comment:
Pendant le chargement la sidebar affiche : 
![selection_205](https://user-images.githubusercontent.com/1556356/36081311-23b5cd8c-0ff1-11e8-89c5-8a63ad1d37c0.png)

Il faut un moyen de juste masquer totalement un item sans message si c'est spécifié

Reply:
j'ai fait : 
```
            {this.props.canAccessModeration ? <this.MenuListLink to="/moderation" iconName="flag" className="hide-when-collapsed">
              { t('menu.moderation') }
```

Au début je voulais ajouter le controle de la réputation requise au composant MenuLink. Mais il n'a pas accès au store, d'ailleurs je trouve ca bizarre ta façon de déclarer des composants "internes" je trouve ca pas clair la façon dont l'objet this est passé au fils

Reply:
Ton dumb component peut être une simple fonction qui retourne un composant. Ici c'est le cas, 

```jsx
<this.MenuListLink x="42"/>
```

Sera compilé par React en : 

```javascript
React.createElement(this.MenuListLink(), {x: "42"});
```

L'objet `this` n'est pas directement passé au fils, c'est juste que c'est une méthode et pas une fonction. 

Sinon plutôt que dupliquer le code de contrôle de la vérification, tu peux accepter un paramètre `hideWhenNotAllowed` dans le ReputationGuard et retourner `null` quand il n'y a pas assez de réputation.

Reply:
Voilà c'est fait